app/scripts/controllers/edit/buildConfig.js

```diff
@@ -154,89 +130,53 @@ angular.module('openshiftConsole')
             _.each($scope.envVars, function(env) {
               $filter('altTextForValueFrom')(env);
             });
-            $scope.triggers = $scope.getTriggerMap($scope.triggers, $scope.buildConfig.spec.triggers);
-            $scope.sources = $scope.getSourceMap($scope.sources, $scope.buildConfig.spec.source);
+            $scope.triggers = getTriggerMap($scope.triggers, $scope.buildConfig.spec.triggers);
+            $scope.sources = getSourceMap($scope.sources, $scope.buildConfig.spec.source);
 
             if (_.has(buildConfig, 'spec.strategy.jenkinsPipelineStrategy.jenkinsfile')) {
               $scope.jenkinsfileOptions.type = 'inline';
             }
 
-            if ($scope.buildStrategy.from) {
-              var buildFrom = $scope.buildStrategy.from;
-              $scope.builderOptions = $scope.setPickedVariables(
-                $scope.builderOptions,
-                buildFrom.kind,
-                buildFrom.namespace || buildConfig.metadata.namespace,
-                buildFrom.name.split(":")[0],
-                buildFrom.name.split(":")[1],
-                (buildFrom.kind === "ImageStreamImage") ? buildFrom.name : "",
-                (buildFrom.kind === "ImageStreamTag") ? buildConfig.metadata.namespace + "/" + buildFrom.name : buildFrom.name);
-            } else {
-              $scope.builderOptions = $scope.setPickedVariables($scope.builderOptions, "None", buildConfig.metadata.namespace, "", "", "", "");
-            }
+            var setImageOptions = function(imageOptions, imageData) {
+              imageOptions.type = (imageData && imageData.kind) ? imageData.kind : "None";
+              var ist = {},
+                  isi = "",
+                  di = "";
```

Comment:
These names are a bit too short for me. Maybe `istag`, `isimage`, and `dockerImage`? `istag` and `isimage` then match the CLI abbreviations.